src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -812,8 +831,157 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
+
+        while status_file_size_in_bytes > Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES:
+            # Start truncation process
+            packages_retained_in_assessment, packages_removed_from_assessment, packages_retained_in_installation, packages_removed_from_installation = \
+                self.__start_truncation_process(self.__assessment_packages_copy, self.__installation_packages_copy, size_of_max_packages_allowed_in_status, low_pri_index)
+
+            if len(packages_removed_from_assessment) > 0:
+                self.__assessment_packages_removed = packages_removed_from_assessment
+                self.composite_logger.log_debug("Recomposing truncated status payload: [Substatus={0}]".format(Constants.PATCH_ASSESSMENT_SUMMARY))
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file=truncated_status_file, truncated_package_list=packages_retained_in_assessment,
+                    substatus_message=self.__assessment_substatus_msg_copy, substatus_index=assessment_substatus_index)
+
+            if len(packages_removed_from_installation) > 0:
+                self.__installation_packages_removed = packages_removed_from_installation
+                self.composite_logger.log_debug("Recomposing truncated status payload: [Substatus={0}]".format(Constants.PATCH_INSTALLATION_SUMMARY))
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file=truncated_status_file, truncated_package_list=packages_retained_in_installation,
+                    substatus_message=self.__installation_substatus_msg_copy, substatus_index=installation_substatus_index)
+
+            status_file_size_in_bytes = self.__calc_status_size_on_disk(json.dumps(truncated_status_file))
+            status_file_agent_size_diff = status_file_size_in_bytes - Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES
+            size_of_max_packages_allowed_in_status -= status_file_agent_size_diff   # Reduce the max packages byte size by tombstone, new error, and escape chars byte size
+
+        self.composite_logger.log_verbose("End package list truncation")
+        return truncated_status_file
+
+    def __split_assessment_list(self, assessment_packages):
+        """ Split assessment list into 2 parts. MIN_ASSESSMENT_PACKAGE_TO_RETAIN count as one part and remaining as the other """
+        min_packages_count = Constants.StatusTruncationConfig.MIN_ASSESSMENT_PACKAGE_TO_RETAIN
+        min_assessment_patches_to_retain, remaining_assessment_patches = (assessment_packages[:min_packages_count], assessment_packages[min_packages_count:]) \
+            if len(assessment_packages) > min_packages_count else (assessment_packages, [])
+        return min_assessment_patches_to_retain, remaining_assessment_patches
+
+    def __start_truncation_process(self, assessment_packages, installation_packages, max_package_list_byte_size, low_pri_index=None):
```

Comment:
max_package_list_size_in_bytes or max_patch_list_size_in_bytes

Reply:
Looking further at the implementation, rename this to max_allowed_package_list_size_in_bytes or 
 max_allowed_patch_list_size_in_bytes

Reply:
rename related vars to end with in_bytes, done